src/engine/ScreenFilter.cs

```diff
@@ -0,0 +1,37 @@
+﻿using Godot;
+
+public class ScreenFilter : ColorRect
+{
+    private ShaderMaterial? material;
+
+    public override void _EnterTree()
+    {
+        base._EnterTree();
+
+        material ??= (ShaderMaterial)Material;
+
+        UpdateEffect(Settings.Instance.CurrentScreenEffect);
+
+        Settings.Instance.CurrentScreenEffect.OnChanged += UpdateEffect;
+    }
+
+    public override void _ExitTree()
+    {
+        base._ExitTree();
+
+        Settings.Instance.CurrentScreenEffect.OnChanged -= UpdateEffect;
+    }
+
+    public void UpdateEffect(ScreenEffect? currentEffect)
+    {
+        if (currentEffect?.ShaderPath == null)
+        {
+            ((ShaderMaterial)Material).Shader = null;
```

Comment:
Why does this not use the `material` variable you setup in _EnterTree?

Reply:
It doesn't like it, as material might be null. Cleaned it up.

Reply:
Well the opposite fix should be done. It's a pattern used everywhere else that you take the resources once when starting and then use them. If something is not retrieved yet / null when it shouldn't be throw an InvalidOperationException.